src/server/main.ts

```diff
@@ -25,17 +28,25 @@ app.post('/register', async (req, res) => {
   if (!userExists) {
```

Comment:
now that you have UserService, I would have a `register` method on your UserService class which calls the database service